pkgs/applications/misc/ethereum/cpp/default.nix

```diff
@@ -0,0 +1,134 @@
+{ stdenv
+, fetchgit
+, cmake
+, boost
+, gmp
+, jsoncpp
+, leveldb
+, cryptopp
+, cudatoolkit
+, nvidia_x11
+, libcpuid
+, miniupnpc
+, libjson_rpc_cpp
+, curl
+, libmicrohttpd
+, mesa
+, qtwebengine ? null
+, qtbase ? null
+, qtdeclarative ? null
+}:
+let
+  withGUI = (qtwebengine != null) && (qtbase != null) && (qtdeclarative != null);
+in
+stdenv.mkDerivation rec {
+  name = "cpp-ethereum-${version}";
+  version = "1.2.9";
+
+  src = fetchgit {
+    url = https://github.com/ethereum/webthree-umbrella.git;
```

Comment:
*Ideally* we could separate each submodule into it's own package. They seem to like to split modules up for no apparent reason, so it's not really a big deal. For instance, I don't think you'd ever want "webthree-helpers" but not "libweb3core". These dependency graphs certainly don't lessen my theory that Ethereum is actually crypto Ponzi scheme designed to be so complicated the devs can intentionally insert vulnerabilities for their own financial gain. But, nevermind that :)

https://camo.githubusercontent.com/b59d6b4aa624edfa51c707ac543369c1d90b2e3e/687474703a2f2f646f75626c657468696e6b636f2e6769746875622e696f2f6370702d657468657265756d2d63726f73732f696d616765732f646570656e64656e63795f67726170682e737667